src/feedlistformaction.cpp

```diff
@@ -423,30 +423,24 @@ bool FeedListFormAction::process_operation(Operation op,
 	}
 	break;
 	case OP_SELECTFILTER:
-		if (filters.size() > 0) {
-			std::string newfilter;
+		if (filter_container.size() > 0) {
 			if (automatic && args->size() > 0) {
-				newfilter = (*args)[0];
-			} else {
-				newfilter = v->select_filter(
-						filters.get_filters());
-			}
-			if (newfilter != "") {
-				filterhistory.add_line(newfilter);
-				if (newfilter.length() > 0) {
-					if (!matcher.parse(newfilter)) {
-						v->get_statusline().show_error(strprintf::fmt(
-								_("Error: couldn't "
-									"parse filter "
-									"expression `%s': %s"),
-								newfilter,
-								matcher.get_parse_error()));
-					} else {
-						save_filterpos();
-						apply_filter = true;
-						do_redraw = true;
-					}
+				const std::string filter_name = (*args)[0];
+				const auto filters = filter_container.get_filters();
+				const auto filter = std::find_if(filters.begin(),
+				filters.end(), [&](const FilterNameExprPair& pair) {
+					return pair.name == filter_name;
+				});
```

Comment:
Could this be turned into a method on `FilterContainer`?

Reply:
Jup, done: https://github.com/newsboat/newsboat/pull/2137/commits/cda2a9d4a892bb7129e9eb0c8b883a2b0cb1925c